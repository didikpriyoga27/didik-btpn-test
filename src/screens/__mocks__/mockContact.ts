import {Contact} from '../../slices/contactSlice';

export const mockContact: Contact = {
  age: 20,
  firstName: 'John',
  lastName: 'Doe',
  id: 'id-1',
  photo:
    'https://ui-avatars.com/api/?name=John+Doe&background=0f766e&color=fff',
};

export const mockContact2: Contact = {
  ...mockContact,
  firstName: 'Ahmad',
  id: 'id-2',
};

export const mockContact3: Contact = {
  ...mockContact,
  firstName: 'Zayn',
  id: 'id-3',
};

export const mockContact4: Contact = {
  ...mockContact,
  firstName: 'John',
  id: 'id-4',
};
