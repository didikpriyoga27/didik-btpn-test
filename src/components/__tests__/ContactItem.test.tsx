import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import {mockContact} from '../../screens/__mocks__/mockContact';
import ContactItem from '../ContactItem';

jest.useRealTimers();

describe('@components/ContactItem', () => {
  const component = () => render(<ContactItem item={mockContact} />);

  test('snapshot: component', async () => {
    const {toJSON, getByTestId} = component();

    const ButtonContactDetail = getByTestId('ButtonContactDetail');
    const ButtonDeleteContact = getByTestId('ButtonDeleteContact');

    fireEvent(ButtonContactDetail, 'press');
    fireEvent(ButtonDeleteContact, 'press');

    expect(toJSON()).toMatchSnapshot();
  });
});
