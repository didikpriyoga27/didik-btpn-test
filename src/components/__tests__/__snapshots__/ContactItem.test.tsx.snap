// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/ContactItem snapshot: component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 16,
      "opacity": 1,
    }
  }
  testID="ButtonContactDetail"
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
        ],
      ]
    }
  >
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          [
            {
              "width": 48,
            },
            {
              "height": 48,
            },
            {
              "marginRight": 16,
            },
            {
              "borderBottomLeftRadius": 9999,
              "borderBottomRightRadius": 9999,
              "borderTopLeftRadius": 9999,
              "borderTopRightRadius": 9999,
            },
            {
              "backgroundColor": "#d1d5db",
            },
          ],
        ]
      }
    >
      <FastImageView
        defaultSource={null}
        resizeMode="cover"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
          },
          [
            {
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
            },
            {
              "flexDirection": "row",
            },
            {
              "alignItems": "center",
            },
            {
              "borderBottomWidth": 1,
            },
            {
              "borderBottomColor": "#d1d5db",
              "borderLeftColor": "#d1d5db",
              "borderRightColor": "#d1d5db",
              "borderTopColor": "#d1d5db",
            },
            {
              "paddingBottom": 12,
            },
          ],
        ]
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#fff",
            },
            [
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ],
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#000",
              },
              [
                {
                  "fontWeight": "700",
                },
              ],
            ]
          }
        >
          John
           
          Doe
        </Text>
        <Text
          style={
            [
              {
                "color": "#000",
              },
            ]
          }
        >
          20
           Years
        </Text>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#fff",
            },
            [
              {
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              },
              {
                "width": 32,
              },
              {
                "height": 32,
              },
              {
                "alignItems": "center",
              },
              {
                "justifyContent": "center",
              },
            ],
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="ButtonDeleteContact"
        >
          <RNSVGSvgView
            bbHeight={25}
            bbWidth={25}
            fill="none"
            focusable={false}
            height={25}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 25,
                  "width": 25,
                },
              ]
            }
            width={25}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M17.338 6.525h5v2h-2v13a1 1 0 0 1-1 1h-14a1 1 0 0 1-1-1v-13h-2v-2h5v-3a1 1 0 0 1 1-1h8a1 1 0 0 1 1 1v3Zm1 2h-12v12h12v-12Zm-4.586 6 1.768 1.768-1.414 1.414-1.768-1.768-1.768 1.768-1.414-1.415 1.768-1.767-1.768-1.768 1.414-1.414 1.768 1.768 1.768-1.768 1.414 1.414-1.768 1.768Zm-4.414-10v2h6v-2h-6Z"
                fill={
                  {
                    "payload": 4290321436,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
</View>
`;
